ls1intum/Artemis
# `Programming exercises`: Add support for testwise coverage analysis for Java/Kotlin exercises on Jenkins setup

### Checklist
#### General
#### Server
#### Client
#### Changes affecting Programming Exercises
_not applicable_

### Motivation and Context
In #4889 and #5220, support has been added to perform a testwise coverage analysis for Java/Kotlin exercises on the Atlassian Setup. These changes add the functionality for the Jenkins-setup.

### Description
The pipeline scripts have been adjusted, as well as the DTOs used for the test results.

### Steps for Testing
Prerequisites:
- 1 Instructor on TS 2

1. Log in to Artemis
2. Create a new programming exercise with the following configuration:
- Programming Language: Java
- Project Type: Maven
- SCA: enabled
3. Wait until the solution build plan has finished.
4. Click the button "Show testwise coverage" from the programming exercise details page.
5. Make sure that the opened model looks as in #4889.
6. Repeat step 2 - 5 for Java-Gradle and Kotlin (without SCA).

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
_no changes_
